src/wallet/wallet.cpp

```diff
@@ -3843,9 +3847,9 @@ CWallet* CWallet::CreateWalletFromFile(const std::string walletFile)
         walletInstance->SetBestChain(chainActive.GetLocator());
     }
     else if (gArgs.IsArgSet("-usehd")) {
```

Comment:
Instead of moving the code block up, couldn't this just be made an `if` instead of `else if`? If we want to show this error before doing any of the wallet creating stuff, this entire check could just be moved to before the first run block.

Reply:
Yes, the else can be removed for better clarity. However, I'd still prefer that the new-non-hd-error is guarded by first-run. This way we can return early and don't have to deal with `TopUpKeyPool`, which might take a "long" time...

I don't think this whole check can be moved to before the first run block, as `IsHDEnabled` should be called after (a potential) `SetHDMasterKey`.